cmds/core/uptime/uptime_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"testing"
+	"time"
+)
+
+type UptimeOutput struct {
+	uptime time.Time
+	err    error
+}
+
+type LoadAvgOutput struct {
+	loadavg string
+	err     error
+}
+
+func TestUptimeGoodData(t *testing.T) {
+	time := time.Date(0001, 1, 15, 5, 35, 49, 0, time.UTC)
+	want := UptimeOutput{time, nil}
+	uptime, err := uptime("1229749 1422244")
+	got := UptimeOutput{*uptime, err}
+	if !got.uptime.Equal(want.uptime) {
+		t.Errorf("the uptime got is not correct:want %v , got %v", want.uptime, got.uptime)
+	}
+	if got.err != want.err {
+		t.Errorf("expected no error from a good proc/uptime file but got this error:%v", got.err)
+	}
+}
+
+func TestUptimeBadData(t *testing.T) {
+	time := time.Date(0000, 0, 0, 0, 0, 0, 0, time.UTC)
+	_, err := uptime("string")
+	got := UptimeOutput{time, err}
+	want := UptimeOutput{time, errors.New("could not get uptime:error parsing the string uptime to a time duration")}
+	if got.err.Error() != want.err.Error() {
+		t.Errorf("Error wanted: %v , Error got : %v", want.err, got.err)
+	}
+}
+
+func TestUptimeEmptyData(t *testing.T) {
+	time := time.Date(0000, 0, 0, 0, 0, 0, 0, time.UTC)
+	_, err := uptime(" ")
+	got := UptimeOutput{time, err}
+	want := UptimeOutput{time, errors.New("the contents of proc/uptime we are trying to read are empty")}
```

Comment:
You don't need to make this an error. You can compare the string to got.err.Error(). Error() returns the string.

Reply:
Likewise for all the other error checks in this file

Reply:
Hey Gan .I think using using errors in this case is the most efficient and easily understandable way to implement the uptime_test.go. I tried to make the errors strings but that brought in a whole bunch of problems.For example ,when the return error is nil and you try to use err.Error() the whole code seg faults.What is the reason behind using strings?

Reply:
Wait, if the return error is nil, you still have a problem when you use got.err.Error(). Wrapping the expect in a struct does nothing, and converting your expected error string to an error type doesn't solve this problem. You can special case this right now when you know it won't exist, but you can't do it when you move to table driven tests.

Reply:
Done